### ISSUE TITLE
Table image in column

### ISSUE THREAD

**Issue by GeorgeD19** [jobsta/reportbro-designer#12]
Is there any way on inserting an image into a table column?

**Comment by alhman** [comment]
Right now a table is only meant for simple tabular text display. If you need more complex layouts you can use sections which are more flexible.

**Comment by rakesh-ksolves** [comment]
@alhman - is there any way to add image dynamically in the report ? or is there any way to render the image of richTextArea field?

**Comment by 4t4nner** [comment]
How add various images with data when I generate report?

```python
# data.toImageSectionData[0].image is a base64 image
report = Report(report_definition, data, is_test_data, additional_fonts=additional_fonts)
```

when I add image to template, get something like [this](https://i.imgur.com/YgIX3Ds.png)

**Comment by alhman** [comment]
you can pass images as base64 encoded or I/O object. See data parameter in the [method documentation](https://www.reportbro.com/docs/methods#reportbro-lib)

**Comment by 4t4nner** [comment]
> you can pass images as base64 encoded or I/O object

What type should I use for it in frontend? List type contains only simple types. I need various picture for each row. 

![2020-04-08_15-46](https://user-images.githubusercontent.com/28292794/78764005-20387f80-79b0-11ea-9344-8adff707b00a.png)

**Comment by alhman** [comment]
you're right, Image type should be allowed in a list. We will add this in the next version.

With a text parameter used for an image you can pass a url (starting with "https://" or "http://"). In the next version we'll also allow specifying a file path (starting with "file:" ).

**Comment by lachlanhickey** [comment]
> With a text parameter used for an image you can pass a url (starting with "https://" or "http://"). In the next version we'll also allow specifying a file path (starting with "file:" ).

Is it possible to have the image specified via a URL load within the builder? Ie, i wish to see the image whilst I'm editing

**Comment by alhman** [comment]
there is no preview of image links in the Designer. You could download the file and select it for the image file. Then you get a preview and for rendering the image link will be used.

**Comment by lachlanhickey** [comment]
Thanks for your response. Is there a way to integrate the loading of an image via a URL?

It does make it difficult to design a report around an image that is only seen as a box. A scenario would be a dynamic background image that is loaded, that has placeholder spaces. I need to be able to place (on top of the image/background) certain fields.

**Comment by alhman** [comment]
image preview for images specified by url is now supported

**Comment by lachlanhickey** [comment]
Thank you for all these new features/bugfixes. These are fantastic!!

**Comment by lachlanhickey** [comment]
Further to the above, should I be able to have an image, with the 'Source' being from a dynamic parameter such as a URL?

**Comment by alhman** [comment]
you can have a parameter (type: text) which contains the url and an image with the source containing the parameter. The image will be shown when the report is generated. The preview does only work in case the url is directly set as image source and not by parameter because the parameter is evaluated at runtime.

So I guess that's still a problem in your case? We'll see if we can use the parameter test data for image preview in case a parameter is used.

**Comment by lachlanhickey** [comment]
The preview is not an issue in this instance. 

However, I think I found a further 'bug'. The suggested function only works if it is not a child object. 

Only top-level parameters seem to work in the suggested manner. If I am using a child as the image source I get a 'Parameter not found' error.

**Comment by alhman** [comment]
You're right, when the parameter (containing the image url) used for image source belongs to a collection parameter it doesn't work. We'll try to fix this with the next update.

**Comment by alhman** [comment]
Using a parameter with image url inside a collection is now possible with the latest release 1.5.0

**Comment by mkusy** [comment]
Hello! I try to display an image in the table column with following steps:

1. I've added a Table to my report.
2. In the Designer, I've added a Parameter - the List corresponding to the table data source.
3. I've set the type of one of the parameters from the list to Image.
4. I've set the corresponding column's Text to use the parameter mentioned above. 
5. I send the generation request to my backend where I set the value of the parameter (3) to base64-encoded image data accordingly.
6. I forward the request to the ReportBro server.

As a result, I would expect my images to be rendered in the table column/rows. Unfortunately, I see that my cell contains the base64-encoded image data displayed as a string.

What am I doing wrong?

**Comment by alhman** [comment]
You should use a section because images in tables are not supported.
